### PR TITLE
remove onBlur and handle mousedown outside

### PR DESCRIPTION
**Pull request by kratuwu**

### CODE REVIEW

File: src/components/PartyDropdown.js [hunk 1]
```diff
@@ -11,12 +11,24 @@ const searcher = new Fuse(parties, {
...
 
+  const dropdownRef = useRef()
+
+  const handleClickOutside = e => {
+    if (!!dropdownRef.current && !dropdownRef.current.contains(e.target)) {
+      setDropdownOpen(false)
+    }
+  }
+  useEffect(() => {
+    document.addEventListener("mousedown", handleClickOutside)
```
Comment (dtinth) [inline src/components/PartyDropdown.js:29]: useEffect should return an unsubscribe function krub, otherwise there will be memory leak.

Comment (kratuwu) [reply src/components/PartyDropdown.js:29]: fixed krub

**Comment by p16i** [comment]
just out of curiosity, why should we use `mousedown` instead of `onBlur` in this case ?

**Comment by kratuwu** [comment]
If use onBlur, it'll can't change partyPath when select dropdown.

File: src/components/PartyDropdown.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ export default ({ partyId }) => {
   }
   useEffect(() => {
     document.addEventListener("mousedown", handleClickOutside)
+    return () => document.addEventListener("mousedown", handleClickOutside)
```
Comment (p16i) [inline src/components/PartyDropdown.js]: should this line be `document.removeEventListener`?

Comment (kratuwu) [reply src/components/PartyDropdown.js]: fixed krub

File: src/pages/party.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { DISPLAY_FONT } from "../styles"
 
 export default ({ pageContext, location }) => {
   const partyId = pageContext.partyId
+  console.log(partyId)
```
Comment (p16i) [inline src/pages/party.js]: please remove this line.

Comment (kratuwu) [reply src/pages/party.js]: Removed krub

Comment (p16i) [reply src/pages/party.js]: thanks!

**Pull request merged**